vnext/docs/NativeModules.md

```diff
@@ -0,0 +1,54 @@
+# Native Modules Guide for React Native Windows (vnext)
+
+React Native supports interop between JavaScript and native code through the use of native modules.
+
+Custom native modules provide methods and constants implemented in native language that are exposed for use in JavaScript. For example, if you needed access to a thrid-party native image processing library, you could implement a custom native module as a wrapper to expose that library's code to your React Native app.
+
+Custom view managers are specialized native modules that expose native UI controls for use as React Native components. For example, if you wanted to use a third-party XAML control in your React Native app you would create a custom view manager to register that control as a component that you could use then in your React Native app.
+
+## Current Support
+
+| UWP Host Application | Custom Native Modules | Custom View Managers |
+|:---------------------|:---------------------:|:--------------------:|
+| C++/WinRT            | Yes (C++/WinRT only)  | Yes (C++/WinRT only) |
+| C#, C++/CX           | Yes (C#, C++/CX only) | Not yet              |
+
+
+## High-Level Overview
+
+### C++/WinRT UWP Host Application
+
+#### Creating a Custom Native Module
+
+1. Subclass `facebook::xplat::module::CxxModule`
+  1. Override `getName()` with your module's name
+  2. Override `getMethods()` with the list of methods your module will provide
```

Comment:
we will eventually need to document getMethods with examples, they are tricky - especially with promises (2 callbacks) there is a requirement that the result you return in the failure callback follows a specific json format